src/components/badges/BadgeTitle.tsx

```diff
@@ -1,26 +1,24 @@
-import BaseProof from 'helpers/BaseProof'
-import ContractName from 'components/ContractName'
-import ERC721Proof from 'helpers/ERC721Proof'
-import EmailProof from 'helpers/EmailProof'
-import ExternalLink from 'components/ExternalLink'
+import { isAddress } from 'ethers/lib/utils'
+import ContractName from 'components/ui/ContractName'
+import ExternalLink from 'components/ui/ExternalLink'
 import Network from 'models/Network'
-import getEtherscanAddressUrl from 'helpers/getEtherscanAddressUrl'
+import Proof from 'models/Proof'
+import getEtherscanAddressUrl from 'helpers/network/getEtherscanAddressUrl'
 
-function ProofName({ badge }: { badge: BaseProof }) {
-  if (badge instanceof ERC721Proof)
+function ProofName({ proof }: { proof: Proof }) {
+  if (isAddress(proof.original))
```

Comment:
You're using `proof.original` two times, maybe it's better to destructure it on the top?